### PR TITLE
Unit Test suites for AsyncTasks

### PR DESCRIPTION
**Pull request by sanyud24**
Tested custom AsyncTask classes.
### CheckpointResolutionListTask 
- Used Roboelectric `ShadowEnvironment.setExternalStorageState(Environment.MEDIA_MOUNTED)` to change the 
   "card-status = mounted" and to set mock external storage.
- Mockito to mock ResolutionListener class

### CODE REVIEW

File: services_app/src/test/java/org/opendatakit/services/resolve/CheckpointResolutionListTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public void checkIfRowEntryAdapter_isResolved() {
+        checkpointResolutionListTask.setResolveRowEntryAdapter(adapter);
+        assertEquals(adapter.getCount(), checkpointResolutionListTask.getResolveRowEntryAdapter().getCount());
+        mProgress = String.format(RESOLVING_ROWS, 1, adapter.getCount());
+        assertEquals(mProgress, checkpointResolutionListTask.getProgress());
+
+    }
+
+    @Test
+    public void checkIfResolutionListener_isAvailable() {
```
Comment (wbrunette) [inline services_app/src/test/java/org/opendatakit/services/resolve/CheckpointResolutionListTaskTest.java:67]: This test does not seem to verify any behavior. Normally you see an assert to verify it's working

**Review by wbrunette — CHANGES_REQUESTED** [review]
Your setup creates a background task each time. Does the task run? What happens if you cancel the task? What about making sure the task finishes?

**Comment by sanyud24** [comment]
> Your setup creates a background task each time. Does the task run? What happens if you cancel the task? What about making sure the task finishes?

Yeah, ur right!! added the asynctask.execute to make sure task runs and asynctask.cancel to make sure asynctask thread stops.

**Comment by sanyud24** [comment]
Hi @wbrunette, There's still the error of OdkConnectionFactorySingleton not initialized as it's a final class and mockito can't mock it. I'll look into the issue.

**Comment by wbrunette** [comment]
@sanyud24 did you solve the issue on of the mockito?

**Comment by sanyud24** [comment]
> @sanyud24 did you solve the issue on of the mockito?

Yeah @wbrunette , I'm almost done with it.

**Comment by sanyud24** [comment]
Hi, @wbrunette fixed this PR and added a base test class.

**Review by wbrunette — APPROVED** [review]

**Pull request merged**